tree/treeplayer/test/dataframe/datasource_trivial.cxx

```diff
@@ -98,3 +98,39 @@ TEST(TTrivialDS, FromATDFWithJitting)
    EXPECT_DOUBLE_EQ(9., *max);
    EXPECT_DOUBLE_EQ(22., *min);
 }
+
+// NOW MT!-------------
+
+TEST(TTrivialDS, FromATDFMT)
+{
+   ROOT::EnableImplicitMT(4);
+   std::unique_ptr<TDataSource> tds(new TTrivialDS(320));
+   TDataFrame tdf(std::move(tds));
+   auto max = tdf.Max<ULong64_t>("col0");
+   auto min = tdf.Min<ULong64_t>("col0");
+   auto c = tdf.Count();
+   auto max2 = tdf.Filter([](ULong64_t col0) { return col0 < 10; }, {"col0"}).Max<ULong64_t>("col0");
+   auto min2 = tdf.Filter([](ULong64_t col0) { return col0 > 10; }, {"col0"})
+                  .Define("j", [](ULong64_t col0) { return col0 * 2; }, {"col0"})
+                  .Min<ULong64_t>("j");
+
+   EXPECT_EQ(320U, *c);
+   EXPECT_DOUBLE_EQ(319., *max);
+   EXPECT_DOUBLE_EQ(0., *min);
+   EXPECT_DOUBLE_EQ(9., *max2);
+   EXPECT_DOUBLE_EQ(22., *min2);
```

Comment:
Why `EXPECT_DOUBLE_EQ` on `long`s?

Reply:
Unfortunately at the moment `Min` and `Max` always return doubles.

Ideally we would return the exact column type for calls like `Max<T>("x")` and double for jitted calls, `Max("x")`, but I never had the time to implement the required logic.

Reply:
I think it's low priority at the moment to iron out this one...

Reply:
Here is an example of why I think this is a bad idea:
```
$ root -l
root [0] x = 10101010101010101010.0;
root [1] x
(double) 10101010101010100224.000000
root [2] 
```

Reply:
Yes I agree it is a missing feature

Reply:
To put it in other words, ints/longs are exact, converting them to doubles will break that in legitimate situations when numbers are large enough.

Reply:
Icannot but keep agreeing :P PRs are welcome! :)